src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -275,6 +302,7 @@ protected FileMonitoringController(FilePath ws) throws IOException, InterruptedE
             } else {
                 controlDir = null;
             }
+            cookieValue = cookieFor(ws);
```

Comment:
This should rather be a constructor parameter, defaulting like this (to the _old_ format) only in the case that the implementation of `launchWithCookie` or `doLaunch` neglected to use the new constructor overload. (And of course deprecate the original overload and fix the callers in this repo.)

Reply:
IIUC you are proposing:
- to deprecate the current `FileMonitoringController` constructor and there set up `cookieValue` to the old format.
- to create a new one with `cookieValue` as constructor parameter (value that could be `null` or being provided by the callers - expected the new format)
- to modify all the callers to use the new one

It sounds to me like introducing some extra complexity, and, in addition, making it feasible in code to still have the old format on launching new tasks. Idea was to replace/switch and maintain backward compatibility just for previously launched tasks on nodes with restart capabilities.

Reply:
> value that could be `null`

No, non-null.

> introducing some extra complexity

Not much I think, just threading a variable through another call, rather than trying to recompute it and hoping you come up with the same result.

> still have the old format on launching new tasks

No, because the impls of `DurableTask` in this plugin (which is probably all of them in the world) are updated to call the new constructor.

Reply:
Got it, refactored in https://github.com/jenkinsci/durable-task-plugin/pull/141/commits/2fb15cccaba6b6fcd4594628b065eff267b7b43d